lib/virtual-net/src/host.rs

```diff
@@ -132,6 +166,13 @@ impl VirtualNetworking for LocalNetworking {
         port: Option<u16>,
         dns_server: Option<IpAddr>,
     ) -> Result<Vec<IpAddr>> {
+        if let Some(ruleset) = self.ruleset.as_ref() {
+            if !ruleset.matches_domain(host) {
+                tracing::warn!(%host, "ruleset matching failed");
+                return Err(NetworkError::PermissionDenied);
+            }
```

Comment:
The resolved IPs have to be added to the whitelist.

Reply:
Done